subgraph/src/fleek-nfa.ts

```diff
@@ -257,6 +254,91 @@ export function handleMetadataUpdateWithIntValue(
   }
 }
 
+export function handleInitialized(event: InitializedEvent): void {
+    // This is the contract creation transaction.
+    log.warning('This is the contract creation transaction.', []);
+    if (event.receipt) {
+      let receipt = event.receipt as ethereum.TransactionReceipt;
+      log.warning('Contract address is: {}', [
+        receipt.contractAddress.toHexString(),
+      ]);
+ 
+      // add owner
+      let owner = new Owner(event.transaction.from);
+      owner.collection = true;
+      owner.save();
+    }
+}
+
+export function handleCollectionRoleChanged(event: CollectionRoleChangedEvent): void {
+  let toAddress = event.params.toAddress;
+  let byAddress = event.params.byAddress;
+  let role = event.params.role;
+  let status = event.params.status;
+  
+  if (role == 0) {
+    // Owner role
+    if (status) {
+      // granted
+      let owner = Owner.load(toAddress);
+      if (!owner) {
+        owner = new Owner(toAddress);
+      }
```

Comment:
```suggestion
      const owner = Owner.load(toAddress) || new Owner(toAddress);
```

Reply:
Discussed with @zoruka on Discord


Summary:

This suggestion fails because of the possibility of null attribution to the owner variable with the loading of the address. The if condition is still necessary.

Not going to be implemented.